chess/src/history.rs

```diff
@@ -37,17 +37,18 @@ impl History {
     }
 
     pub fn add_info(&mut self, next_state: BoardState, mv: Move) {
-        *self.repetition_counter.entry(next_state).or_insert(0) += 1;
-        self.turns.push(Turn::new(next_state, mv));
-        self.current_turn += 1;
-        let previous_board_state = self.get_board_state(self.turns.len() - 1);
-        let is_pawn = previous_board_state.get_piece(&mv.from).unwrap().is_pawn();
-        let is_capture_move = previous_board_state.get_piece(&mv.to).is_some();
+        let current_state = self.get_current_state();
+        let is_pawn = current_state.get_piece(&mv.from).unwrap().is_pawn();
+        let is_capture_move = current_state.get_piece(&mv.to).is_some();
```

Comment:
I think en passant is an exception here.

Reply:
En passant is a capture

Reply:
I know and `get_piece(&mv.to)` would be `None` in that case